From 775222c9216a4baf0970b2bd8ef6b5bb45489311 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tor=20Arne=20Vestb=C3=B8?= <tor.arne.vestbo@qt.io>
Date: Mon, 13 Oct 2025 17:18:02 +0200
Subject: [PATCH 167/239] ScrollBar::test_flashing: Use non-interactive
 scrollbar to stabilize test

In rare cases the mouse was lingering over the exact position of the
scroll bar as a result of other tests moving the mouse, which would
make the scrollbar never leave the active state, and hence keep
reporting a non-0 opacity.

We're programmatically activating the scroll bar via increment(),
so we don't need the scroll bar to be interactive to do the test.

Pick-to: 6.8 6.5
Change-Id: Id75b22d51277c3515fd4f448360bd576b45d69d8
Reviewed-by: Richard Moe Gustavsen <richard.gustavsen@qt.io>
(cherry picked from commit 5d4602a0db9e3c717749cafceb8419048dd2bd58)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 tests/auto/quickcontrols/controls/data/tst_scrollbar.qml | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tests/auto/quickcontrols/controls/data/tst_scrollbar.qml b/tests/auto/quickcontrols/controls/data/tst_scrollbar.qml
index 1495f35f5b..0c6f629f7f 100644
--- a/tests/auto/quickcontrols/controls/data/tst_scrollbar.qml
+++ b/tests/auto/quickcontrols/controls/data/tst_scrollbar.qml
@@ -840,7 +840,9 @@ TestCase {
     }
 
     function test_flashing() {
-        let control = createTemporaryObject(scrollBar, testCase, {size: 0.2})
+        // Use non-interactive scroll bar, to prevent the mouse hovering over the scrollbar
+        // from affecting the test, which may happen in CI as a result of other tests.
+        let control = createTemporaryObject(scrollBar, testCase, {size: 0.2, interactive: false})
         verify(control)
 
         let activeSpy = signalSpy.createObject(control, {target: control, signalName: "activeChanged"})
-- 
2.51.2

